packages/iov-ethereum/src/ethereumconnection.spec.ts

```diff
@@ -84,63 +99,130 @@ describe("EthereumConnection", () => {
     connection.disconnect();
   });
 
-  it("can post transaction", async () => {
-    pendingWithoutEthereum();
+  describe("postTx", () => {
+    let postedTxId: TxId;
+    it("can post transaction", async () => {
+      pendingWithoutEthereum();
+
+      const wallet = Secp256k1HdWallet.fromMnemonic(
+        "oxygen fall sure lava energy veteran enroll frown question detail include maximum",
+      );
+      const mainIdentity = await wallet.createIdentity(HdPaths.bip44(60, 0, 0, 1));
+
+      const recipientAddress = "0xE137f5264b6B528244E1643a2D570b37660B7F14" as Address;
+
+      const sendTx: SendTx = {
+        kind: TransactionKind.Send,
+        chainId: nodeChainId,
+        signer: mainIdentity.pubkey,
+        recipient: recipientAddress,
+        amount: {
+          quantity: "3445500",
+          fractionalDigits: 18,
+          tokenTicker: "ETH" as TokenTicker,
+        },
+        gasPrice: {
+          quantity: gasPrice,
+          fractionalDigits: 18,
+          tokenTicker: "ETH" as TokenTicker,
+        },
+        gasLimit: {
+          quantity: gasLimit,
+          fractionalDigits: 18,
+          tokenTicker: "ETH" as TokenTicker,
+        },
+        memo: "We \u2665 developers – iov.one",
+      };
+      const connection = await EthereumConnection.establish(base);
+      const senderAddress = ethereumCodec.keyToAddress(mainIdentity.pubkey);
+      const query: BcpAccountQuery = { address: senderAddress as Address };
+      const nonceResp = await connection.getNonce(query);
+      const signingJob = ethereumCodec.bytesToSign(sendTx, nonceResp.data[0]);
+      const signature = await wallet.createTransactionSignature(
+        mainIdentity,
+        signingJob.bytes,
+        signingJob.prehashType,
+        nodeChainId,
+      );
+
+      const signedTransaction: SignedTransaction = {
+        transaction: sendTx,
+        primarySignature: {
+          nonce: nonceResp.data[0],
+          pubkey: mainIdentity.pubkey,
+          signature: signature,
+        },
+        otherSignatures: [],
+      };
+      const bytesToPost = ethereumCodec.bytesToPost(signedTransaction);
+
+      const result = await connection.postTx(bytesToPost);
+      postedTxId = result.data.txid;
+      expect(result).toBeTruthy();
+      expect(result.data.message).toBeNull();
+    });
+
+    it("can search previous transaction by hash", async () => {
```

Comment:
I'd move this one to the searchTx because you assume posting works and test the search.

Reply:
done